Dangerfile

```diff
@@ -1,119 +1,161 @@
 # frozen_string_literal: true
 
-# Warn if a pull request is too big
-PR_SIZE = {
-  RECOMMENDED_MAXIMUM: 200,
-  ABSOLUTE_MAXIMUM:    500,
-}.freeze
-
-EXCLUSIONS = [
-  'Gemfile.lock',
-  '.json',
-  'spec/fixtures/',
-  '.txt', '.tsv', '.csv',
-  'spec/support/vcr_cassettes/',
-  'app/swagger',
-  'modules/mobile/docs/'
-].freeze
-
-# takes form {"some/file.rb"=>{:insertions=>4, :deletions=>1}}
-changed_files = git.diff.stats[:files]
-
-excluded_files = changed_files.select { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
-included_files = changed_files.reject { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
-
-# fetch master for diff comparison
-`git fetch origin master`
-
-# get branch name
-current_branch = `git branch --show-current`.chomp
-
-# ignores whitespace for the purpose of determining lines of code changed
-diff_changes = `git diff master...#{current_branch} -w --stat`.split("\n")
-
-if diff_changes.empty?
-  lines_of_code = included_files.sum { |_file, changes| changes[:insertions] + changes[:deletions] }
-else
-  lines_of_code = diff_changes.sum(0) do |change|
-    if change == diff_changes.last || EXCLUSIONS.any? { |exclusion| change.match(exclusion) } || change.match(/\|\s+Bin/)
-      0
-    else
-      change.match(/\|\s+(\d+)/)[1].to_i
+module VSPDanger
+  class Runner
+    def self.run
+      [
+        ChangeLimiter.new.run,
+        MigrationIsolator.new.run
+      ]
     end
   end
-end
 
-if lines_of_code > PR_SIZE[:RECOMMENDED_MAXIMUM]
+  class ChangeLimiter
+    EXCLUSIONS = %w[
+      *.csv *.json *.tsv *.txt Gemfile.lock app/swagger modules/mobile/docs spec/fixtures/ spec/support/vcr_cassettes/
+    ].freeze
+    PR_SIZE = { recommended: 200, maximum: 500 }.freeze
 
-  exclusion_text = <<~EXCLUSIONS_TEXT
+    def run
+      return error if lines_changed > PR_SIZE[:maximum]
+      return warning if lines_changed > PR_SIZE[:recommended]
 
-  #### Exclusions
+      info
+    end
 
-  - #{excluded_files.collect { |key, val| "#{key} (+#{val[:insertions]}/-#{val[:deletions]} )" }.join("\n- ")}
+    private
 
-  ####
+    def error
+      { severity: :error, message: error_message }
```

Comment:
found this ticket we should probably incorporate into this refactor since we're writing new text anyway https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/15715